### PR TITLE
chore: Improve CometExchange metrics

### LINKED ISSUE #625: Improve CometExchange metrics
### What is the problem the feature request solves?

To make it easier to compare Spark and Comet in the Spark UI, I would find it helpful if Comet reported the following metrics:

- shuffle write time
- shuffle bytes written

Screenshot from 2024-07-03 08-57-35

### Describe the potential solution

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by viirya**
## Which issue does this PR close?

Closes #625.

### CODE REVIEW

**Comment by andygrove** [comment]
I ran into an issue trying to test this change with TPC-DS q72:

```
: org.apache.spark.SparkException: Job aborted due to stage failure: Total size of serialized results of 132 tasks (1025.8 MiB) is bigger than spark.driver.maxResultSize (1024.0 MiB)
```

I increased `spark.driver.maxResultSize` to 2GB but then the query failed with:

```
24/08/27 08:02:24 INFO DAGScheduler: ShuffleMapStage 40 (collect at /home/andy/git/apache/datafusion-benchmarks/runners/datafusion-comet/tpcbench.py:87) failed in 23.369 s due to Job aborted due to stage failure: Task 6 in stage 40.0 failed 4 times, most recent failure: Lost task 6.3 in stage 40.0 (TID 374) (192.168.86.42 executor 0): org.apache.comet.CometNativeException: Resources exhausted: Failed to allocate additional 132224 bytes for HashJoinInput[0] with 4809854208 bytes already allocated for this reservation - 68761 bytes remain available for the total pool
```

I then switched to using the Comet 0.2.0-rc1 jar with the same configs and the query ran without error.

**Comment by andygrove** [comment]
I tested with commit `f11679800d2f5461c7b235899ef608584742e8a5` (the upgrade to version 0.3.0) and the query runs without error. I then rebuilt from this PR and I'm back to the query failing with the exact same memory error.

I do not see how the changes in this PR could cause any changes to memory usage though.

**Comment by andygrove** [comment]
Screenshot from 2024-08-27 08-33-11
Screenshot from 2024-08-27 08-33-24

File: native/core/src/execution/datafusion/shuffle_writer.rs [hunk 1]
```diff
@@ -798,7 +810,6 @@ impl ShuffleRepartitioner {
 
     /// Writes buffered shuffled record batches into Arrow IPC bytes.
     async fn shuffle_write(&mut self) -> Result<SendableRecordBatchStream> {
-        let _timer = self.metrics.baseline.elapsed_compute().timer();
         let num_output_partitions = self.num_output_partitions;
         let mut buffered_partitions = self.buffered_partitions.lock().await;
         let mut output_batches: Vec<Vec<u8>> = vec![vec![]; num_output_partitions];
```
Comment (andygrove) [inline native/core/src/execution/datafusion/shuffle_writer.rs:815]: we were previously including the time to run these lines in the total time but now we are not?

Comment (viirya) [reply native/core/src/execution/datafusion/shuffle_writer.rs:815]: I checked how Spark calculates shuffle write time. Spark only includes the time spent on disk operations. I tried to follow it so skipped the lines that are not relevant to disk op.

**Comment by viirya** [comment]
> I ran into an issue trying to test this change with TPC-DS q72:

@andygrove I added `dataSize` back. Could you re-test it? Thanks.

**Comment by andygrove** [comment]
Thanks @viirya. That fixed it. Here are the metric for q72 with Spark and Comet.

Screenshot from 2024-08-28 08-18-41

Screenshot from 2024-08-28 08-18-22

**Review by andygrove — APPROVED** [review]

**Comment by viirya** [comment]
Thank you @andygrove

**Pull request merged**